### ISSUE TITLE
Keyboard position does not follow overlay position

### ISSUE THREAD

**Issue by matejdro** [elvissteinjr/DesktopPlus#31]
With NewUI Preview 3, Keyboard position seems to be stuck in the same place.

I can reproduce with the following steps:

1. Create a new overlay that is positioned to the dashboard
2. Open keyboard on that overlay
3. Close keyboard
4. Spun around 180 degrees
5. Reopen dashboard to reposition new overlay
6. Open keyboard
7. Keyboard will be stuck where it was at step 2

**Comment by elvissteinjr** [comment]
I took a look at it seems to behave correctly on my end. It's not supposed to move when pinned to the play-space and re-positions if it's not and dashboard is open (but also auto-closes if dashboard closed).

Now what I do have notice is that this design is probably not always ideal. My current thoughts about keyboard overlay positioning are the following:

- Dashboard and play-space with separate position/pin state (meaning unpinned in dashboard will always follow the dashboard when it's brought up)
- When pinned from dashboard, also sets pin state and position for play-space
- When pinned from play-space, it only sets pin state for play-space
- When unpinned from play-space, it stays visible and follows overlay it was shown for
- Showing the keyboard in play-space still defaults to pinned

Another potential item on that list would be to keep visibility state for dashboard so window overlays automatically appear again as well when opening the dashboard.

I might be missing some crucial scenario here though, so I'd appreciate it if you have any input before I just rush to implement it like that.

**Comment by matejdro** [comment]
Here is the recording of my issue:

https://user-images.githubusercontent.com/507922/143488861-acd5f48b-dbba-480e-8ac3-0efb98933aed.mp4

**Comment by elvissteinjr** [comment]
Yeah, since the keyboard is pinned in the video it's "working as intended" (has to be pinned, so that happens automatically). The video shows well how wrong these intentions were (what I posted last time were thoughts for changes, not current behavior). Currently, unpinned in Desktop+ dashboard tab follows that, everything else is following play-space, so it won't move on its own beyond initial placement.

What you can do until this is sorted out is to long hold the Show Keyboard button to reset the position. Sorry about that.

**Comment by elvissteinjr** [comment]
I'm cleaning up some old, still open, issues.
Keyboard following overlays it was shown for was added back in NewUI Preview 4, so I assume this can be considered resolved?

If not this can be re-opened, of course.
